core/src/SYCL/Kokkos_SYCL_Instance.hpp

```diff
@@ -136,97 +135,31 @@ class SYCLInternal {
                            Kokkos::Experimental::SYCLDeviceUSMSpace,
                            Kokkos::Experimental::SYCLSharedUSMSpace>;
 
-    // This will memcpy an object T into memory held by this object
-    // returns: a T* to that object
-    //
-    // Note:  it is UB to dereference this pointer with an object that is
-    // not an implicit-lifetime nor trivially-copyable type, but presumably much
-    // faster because we can use USM device memory
-    template <typename T>
-    T* memcpy_from(const T& t) {
-      reserve(sizeof(T));
-      sycl::event memcopied = m_q->memcpy(m_data, std::addressof(t), sizeof(T));
-      fence(memcopied,
-            "Kokkos::Experimental::SYCLInternal::USMObject fence after copy",
-            m_instance_id);
-
-      m_size = sizeof(T);
-      return reinterpret_cast<T*>(m_data);
-    }
-
-    // This will copy-constuct an object T into memory held by this object
-    // returns: a unique_ptr<T, destruct_delete> that will call the
-    // destructor on the type when it goes out of scope.
-    //
-    // Note:  This will not work with USM device memory
-    template <typename T>
-    T* copy_construct_from(const T& t) {
-      static_assert(Kind != sycl::usm::alloc::device,
-                    "Cannot copy construct into USM device memory");
-
-      reserve(sizeof(T));
-
-      m_size = sizeof(T);
-      return new (m_data) T(t);
-    }
-
    public:
-    // Performs either memcpy (for USM device memory) and returns a T*
-    // (but is technically UB when dereferenced on an object that is not
-    // an implicit-lifetime nor trivially-copyable type
-    //
-    // or
-    //
-    // performs copy construction (for other USM memory types) and returns a
-    // reference to the copied object.
+    // Performs either sycl::memcpy (for USM device memory) or std::memcpy
+    // (otherwise) and returns a reference to the copied object.
     template <typename T>
     T& copy_from(const T& t) {
-      fence(m_last_event,
-            "Kokkos::Experimental::SYCLInternal::USMObject fence to wait for "
-            "last event to finish",
-            m_instance_id);
-      m_size = 0;
-      if constexpr (sycl::usm::alloc::device == Kind)
-        return *memcpy_from(t);
-      else
-        return *copy_construct_from(t);
-    }
-
-   private:
-    // Returns a reference to t (helpful when debugging)
-    template <typename T>
-    T& memcpy_to(T& t) {
-      assert(sizeof(T) == m_size);
-
-      sycl::event memcopied = m_q->memcpy(std::addressof(t), m_data, sizeof(T));
-      fence(memcopied,
+      fence();
```

Comment:
No label?

Reply:
I needed to introduce it in #4266 and liked having it even if it's only used in one place. I would not have fought for it if someone didn't want to have it.

Reply:
Well, it is a bit late to request a change